src/Core/Checkout/Customer/Rule/CustomerCustomFieldRule.php

```diff
@@ -160,10 +160,18 @@ private function getValue(array $customFields, array $renderedField)
      */
     private function getExpectedValue($renderedFieldValue, array $renderedField)
     {
-        if (\in_array($renderedField['type'], [CustomFieldTypes::BOOL, CustomFieldTypes::SWITCH], true)) {
+        if ($this->isSwitchOrBoolField($renderedField) && is_string($renderedFieldValue)) {
```

Comment:
What is the reason that requires this change? Is it reproducable, given regular use e.g. in the administration, that the value of a switch or checkbox field gets stored as `string` instead of `bool`? Just asking out of interest.

If we consider all kind of types for the value, the following would also consider values like `1`, `0`, `'1'`, `'0'`, `'yes'`, `'no'`, `'on'` or `'off'`

```php
return filter_var($renderedFieldValue, \FILTER_VALIDATE_BOOLEAN);
```

Reply:
@dneustadt Issue weren't created by me I just searched some easy pick please check https://github.com/shopware/platform/issues/2622 to steps to reproduce :) But yeah it seems like field gets stored as string I would extends tests to cases with '1', '0', 'yes', 'on' etc. and use this filter implementation as it seems like more logical implementation in that case

Reply:
@dneustadt added dataProvider in test for such cases and code updated with suggested implementation with filter_var